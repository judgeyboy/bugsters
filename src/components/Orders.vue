<template>
  <div class="orders">

    <div class="panel">
      <div class="panel-header">
        <div class="panel-title">
          <h2>My Orders</h2>
        </div>
      </div>

      <div class="panel-body">

        <Loader v-if="loading" />

        <OrderTile v-else-if="orders.length > 0"
                   v-for="order in orders"
                   :key="order.id"
                   :order="order" />

        <NoOrders v-else />
      </div>
      <div class="panel-footer">
        <div v-if="orders.length > 0">
          <router-link class="btn btn-primary" to="/">Submit an order</router-link>
        </div>
      </div>
    </div>

  </div>
</template>

<script>
import NoOrders from './NoOrders'
import Loader from './Loader'
import OrderTile from './OrderTile'

export default {
  name: 'Orders',

  data () {
    return {
      loading: false,
      orders: []
    }
  },

  created () {
    this.fetchOrders()
  },

  components: {
    NoOrders,
    Loader,
    OrderTile
  },

  methods: {
    fetchOrders () {
      this.loading = true
      setTimeout(() => {
        this.orders = [
          { id: 'dlkelelege', state: 'in progress' },
          { id: '5e5ge6egeg', state: 'in progress' }
        ]
        this.loading = false
      }, 5000)
    }
  }
}
</script>
