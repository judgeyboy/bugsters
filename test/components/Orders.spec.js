import {shallow} from 'vue-test-utils'

import Orders from '@/components/Orders'

describe('Orders component', () => {
  it('should be a vue instance', () => {
    const wrapper = shallow(Orders)
    expect(wrapper.isVueInstance()).toBe(true)
  })

  it('should match snapshot', () => {
    const wrapper = shallow(Orders)
    const $html = wrapper.vm.$el.outerHTML
    expect($html).toMatchSnapshot()
  })
})
