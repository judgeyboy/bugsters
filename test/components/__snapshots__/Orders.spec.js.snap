// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders component should match snapshot 1`] = `

<div class="orders">
  <div class="panel">
    <div class="panel-header">
      <div class="panel-title">
        <h2>
          My Orders
        </h2>
      </div>
    </div>
    <div class="panel-body">
    </div>
  </div>
</div>

`;
